etc/helm/pachyderm/templates/proxy/service.yaml

```diff
@@ -17,12 +17,22 @@ metadata:
 spec:
   type: {{ .Values.proxy.service.type }}
   ports:
+    {{- if .Values.proxy.service.httpPort }}
```

Comment:
if a user wants TLS and not HTTP, do they have to set proxy.service.httpPort=0 ?

Reply:
Yeah.  Maybe we should document that better.

Reply:
could we just not run this section of template if tls.enabled?

Reply:
It is nice to have HTTP even if you are using TLS because the browser will try port 80 first.  If nothing is listening there, the browser will error before trying HTTPS on port 443.  (I don't know why this is the case.)

What we do when TLS is enabled is make port 80 send a redirect from http://whatever to https://whatever, which upgrades to TLS and starts using port 443 instead.

We do set HSTS ("hypertext strict transport security") headers so that once a user successfully visits pachyderm over https, their browser will prevent them from loading the site over http.  This prevents a man in the middle from downgrading the user to HTTP if they happen to have http://whatever bookmarked.  (I set the caching for this to 1 week, so if a company wants to disable TLS, eventually users will be able to use it without any action.  It's a tradeoff between convenience and security, weighted towards security.  Over on the public Internet, bots are roaming around looking for HSTS headers and they actually ship a list of what they find with the browser.  So if you have never visited, say, jrock.us, and you type http://jrock.us into the URL bar it will never connect to port 80!  This leaves no window for an attacker to MITM the http connection, but of course means I could never turn off HTTPS.)